ua-snap/cmip6-utils
# Prepare regrid pipeline to be used in Prefect

This PR is a companion PR to [PR #22 ](https://github.com/ua-snap/prefect/pull/22)for the `regridding` [branch of the prefect repo](https://github.com/ua-snap/prefect/tree/regridding/regridding).

Some changes were made to the CMIP6 regridding pipeline in order to make it work via Prefect.:

- Instead of using `config.py` and environmental variables, the various directories needed for the pipeline are just constructed from the `{scratch_directory}` parameter in the `regrid_cmip6.regrid_cmip6()` function. (This is similar to what we did in the inidicators flow.) As a result, more arguments have to be passed to `slurm.py` and `regrid.py` for all the directories since they are not readily available from `config.py`.
- A new `run_generate_batch_files.py` script was needed to pass parameters to `generate_batch_files.py`, create the sbatch head and slurm file, and submit a the slurm job for that process.
- The step of submitting the slurm jobs was moved out of the `regrid_cmip6.ipynb` notebook and into the main block of `slurm.py`
- Minor changes to the `no_clobber` parameter. It has to be a string instead of a command line "action". (Not sure of the nomenclature here, but from Prefect we need to provide a string "true"/"false" as the input instead of the inclusion of ```--no_clobber``` as a flag in the original command line version of `regrid.py`. 

**TO TEST:**

- Clone the `regridding` [branch of the prefect repo](https://github.com/ua-snap/prefect/tree/regridding/regridding).
- Review the code changes to `slurm.py` and `regrid.py` and the corresponding arguments passed to tasks in the prefect flow. 
- Review the code in the new `run_generate_batch_files.py` script. Should this be rolled into the `generate_batch_files.py` or is it OK to have that script kind of "wrapped" by this new one?
- Complete the companion prefect PR to run the regridding pipeline on a subset of files, and confirm it works as expected.
